src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -191,4 +192,22 @@ class ReferralService(
   fun getDraftReferralsCreatedByUserID(userID: String): List<DraftReferralDTO> {
     return referralRepository.findByCreatedByIdAndSentAtIsNull(userID).map { DraftReferralDTO.from(it) }
   }
+
+  private fun generateReferenceNumber(category: ServiceCategory): String? {
+    for (i in 1..maxReferenceNumberTries) {
+      val candidate = referenceGenerator.generate(category.name)
+      if (!referralRepository.existsByReferenceNumber(candidate))
+        return candidate
+      else
+        log.warn("Clash found for referral number: $candidate")
+    }
+
+    log.warn("Unable to generate a referral number in $maxReferenceNumberTries tries")
```

Comment:
isn't this an error?

Reply:
Yep, thanks.

Reply:
done in 391573ecc0171576f612823cb30e045fe4f7e7c2